node/full_client_test.go

```diff
@@ -799,7 +803,7 @@ func createGenesisValidators(numNodes int, appCreator func(require *require.Asse
 		err := nodes[i].Start()
 		require.NoError(err)
```

Comment:
We should do a search for these require statements in helpers. 

in this scenario, we should start 3 nodes and fail on the forth, and the first 3 would remaining running. 

We should either just pass in the testing.T and add the stops as `t.Cleanup` functions, or we make this function more like a standard function that just returns an error and we handle the error in the test. 

I'd prefer the later because it offer more flexibility for tests to choose how they stop the nodes if needed.

Reply:
Did a search for it across helpers and made sure there are no instances where `require` would cause us to miss a call to `stop()`. I added the former, `t.Cleanup`, since making it a standard function would require it to change the `appCreator` method which is an argument to it and it makes sense to use `require` inside it.